### PR TITLE
Add Github Actions build and push actions

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

This introduces pushing helm chart and docker image for `main` branch, `release` branches and for tags.
There are 3 separate workflows:

- `PRs checks` - checks for PRs for `main` branch
- `Dev builds` - for `main` and release branches; building docker images, pushing it to docker hub and pushing helm chart to dev repo
- `Release builds` - for release tags; building docker images, pushing it to docker hub and pushing helm chart to main primary repo

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: ci/build.sh [hunk 1]
```diff
@@ -1,166 +1,24 @@
...
-  rm main.tf
-  rm -f ./gems/*.gem
-  popd || exit 1
+if is_checkout_on_tag; then
+  VERSION="$(git describe --tags --always)"
+  readonly VERSION="${VERSION#v}"
+else
+  # Don't use non-tag 'git describe' output as this will
+  # fail when building fluentd plugins.
+  readonly VERSION="0.0.0"
```
Comment (sumo-drosiek) [inline ci/build.sh:14]: Didn't we tested that the plugins are build correctly?

File: ci/build.sh [hunk 2]
```diff
@@ -1,166 +1,24 @@
...
-  echo "Test docker image locally..."
-  ruby deploy/test/test_docker.rb || exit 1
-}
+: "${DOCKER_TAG:=sumologic/kubernetes-fluentd}"
 
-# Set up Github
-if [ -n "${GITHUB_TOKEN}" ]; then
-  set_up_github "${GITHUB_TOKEN}" "$(get_branch_to_checkout)"
-fi
+echo "Starting build in: $(pwd) with version tag: ${VERSION}"
```
Comment (sumo-drosiek) [inline ci/build.sh:19]: Please change `pwd` to the repo main directory. functions won't work otherwise

File: ci/push.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+: "${DOCKER_TAG:=sumologic/kubernetes-fluentd}"
+: "${DOCKER_USERNAME:=sumodocker}"
+
+if [[ -z "${DOCKER_PASSWORD}" ]]; then
+  echo "Provide DOCKER_PASSWORD in order to push built image to container registry"
+  exit 1
+fi
+
+echo "Starting push in: $(pwd) with version tag: ${VERSION}"
```
Comment (sumo-drosiek) [inline ci/push.sh:19]: Please change pwd to the repo main directory. functions won't work otherwise

File: ci/build.sh (hunk 1 above)
Comment (pmalek-sumo) [reply ci/build.sh:14]: Argh, I've removed the old build from GH actions but `git describe ...` returned short SHA which `gem bundle ...` complained about.

File: ci/build.sh (hunk 2 above)
Comment (pmalek-sumo) [reply ci/build.sh:19]: i.e. `cd` to repo root dir or change the `echo`?

Comment (sumo-drosiek) [reply ci/build.sh:19]: `cd`/`pushd` to repo root. `echo` is fine

Comment (sumo-drosiek) [reply ci/build.sh:19]: we are using in functions relative paths, so if you run the script outside of the repo root it's going to fail

File: ci/build.sh (hunk 1 above)
Comment (sumo-drosiek) [reply ci/build.sh:14]: so git describe doesn;t return `1.3.2-1234567`?

File: ci/build.sh (hunk 2 above)
Comment (pmalek-sumo) [reply ci/build.sh:19]: Added `pushd "$(dirname "${BASH_SOURCE[0]}")/.." || exit 1`

File: ci/push.sh (hunk 3 above)
Comment (pmalek-sumo) [reply ci/push.sh:19]: Added `pushd "$(dirname "${BASH_SOURCE[0]}")/.." || exit 1`

File: ci/build.sh (hunk 1 above)
Comment (pmalek-sumo) [reply ci/build.sh:14]: Ah, ok so it didn't but I've added the `git fetch --tags --prune --unshallow` and then I've figured out that we don't really need to fetch full history (via `git fetch --unshallow ...`) but it's enough to fetch current branch (and make sure we have `unshallow`ed repo history

This is handled in `fetch_current_branch` in here: 

https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/5735894346043a74c008733dcb74da34566e6b44/ci/_build_functions.sh#L126-L143

Comment (pmalek-sumo) [reply ci/build.sh:14]: And besides bandwidth we also save time as `git fetch --tags --unshallow` takes roughly 10-20s https://github.com/SumoLogic/sumologic-kubernetes-collection/runs/1464248561?check_suite_focus=true#step:5:1
while fetching only specified branch it takes about a second https://github.com/SumoLogic/sumologic-kubernetes-collection/runs/1470602457?check_suite_focus=true#step:6:5 (till line 72)

File: ci/build.sh [hunk 4]
```diff
@@ -1,166 +1,27 @@
...
+pushd "${ROOT_DIR}" || exit 1
```
Comment (sumo-drosiek) [inline ci/build.sh:20]:
```suggestion
pushd "${ROOT_DIR}" || exit 1
echo "Starting build in: $(pwd) with version tag: ${VERSION}"
```

File: ci/build.sh (hunk 1 above)
Comment (sumo-drosiek) [reply ci/build.sh:14]: So we can simplify the if else statement now

File: ci/tests.sh [hunk 5]
```diff
@@ -2,57 +2,22 @@
...
+source "${ROOT_DIR}"/ci/_build_functions.sh
+
+if is_checkout_on_tag; then
+  VERSION="$(git describe --tags --always)"
+  readonly VERSION="${VERSION#v}"
+else
+  # Don't use non-tag 'git describe' output as this will
+  # fail when building fluentd plugins.
+  readonly VERSION="0.0.0"
+fi
```
Comment (sumo-drosiek) [inline ci/tests.sh:18]: Can we simplify this as well?

Comment (pmalek-sumo) [reply ci/tests.sh]: Done. For the reference: changing this to assign what `git describe --tags` returns will yield in semver versions (e.g. `2.0.0`) for versions that are not a `2.0.0` tag, so `2.0.0-dev.0-136-g205f297` will yields gems with version `2.0.0`

**Review by sumo-drosiek — APPROVED** [review]
LGTM

**Pull request merged**